### ISSUE TITLE
openbsd-7.0-amd64: "Authentication failure"

### ISSUE THREAD

**Issue by dentarg** [twingly/boxes#33]
**tl;dr** I think this has been reported upstream as https://github.com/hashicorp/vagrant/issues/12533

Just tested the newly released `openbsd-7.0-amd64` (https://github.com/twingly/boxes/pull/32) box but Vagrant can't SSH to it:

<Details><Summary><code>vagrant up</code></Summary>

```
$ vagrant up
==> default: Forcing shutdown of VM...
==> default: Destroying VM and associated drives...
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'twingly/openbsd-7.0-amd64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'twingly/openbsd-7.0-amd64' version '1.0.0' is up to date...
==> default: Setting the name of the VM: ansible_default_1635080897762_89372
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 80 (guest) => 8080 (host) (adapter 1)
    default: 53 (guest) => 5353 (host) (adapter 1)
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.
```
</Details>

However, `vagrant provision` (Ansible in my case) can and I can also `ssh` to it using OpenSSH after `vagrant up` has finished (it is up and running despite the `Authentication failure`).

I also ran `vagrant --debug up` (lots of logs!) and I can see the same 

    WARN -- net.ssh.transport.algorithms[c698]: unsupported algorithm: `["rsa-sha2-256", "rsa-sha2-512"]' 

line as https://github.com/hashicorp/vagrant/issues/12533 have.

The server is logging `sshd[97521]: userauth_pubkey: key type ssh-rsa not in PubkeyAcceptedAlgorithms [preauth]` so maybe we can workaround it for now if we modify the `sshd_config` when we build the box.

**Comment by dentarg** [comment]
I think this was addressed in Vagrant 2.3.0 and up: https://github.com/hashicorp/vagrant/blob/v2.3.0/CHANGELOG.md#230-august-5-2022
